File: python/easy/best_time_to_buy_and_sell_stocks_ii.py
class Solution(object):
    import sys
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        if len(prices) <= 1:
            return 0;
        min_price = sys.maxint
        max_profit = 0
        for i in xrange(0, len(prices)-1):
            if prices[i+1] - prices[i] > 0:
                max_profit += (prices[i+1] - prices[i])
        return max_profit

File: cpp/easy/best_time_to_buy_and_sell_stocks_ii.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int max_profit = 0;
        if(prices.size() == 0)
           return max_profit;
        int i = 0;
        for(i = 0; i < prices.size()-1; ++i)
        {
            if(prices[i+1] - prices[i] > 0)
                max_profit += (prices[i+1] - prices[i]);
        }
        return max_profit;
    }
};

Conclusion:
an easy algorithm issue, greedy algorithm.
Time Complexity: O(n)
Space Complexity: O(1)
